Fix macOS CI builds

HofiOne opened the pull request:
`C++ module` now can be built and used on macOS too using gcc-11.
- removed macOS exclusion from `configure`
- modified the Homebrew environment building, added gcc-11, and forced usage of it
- removed the outdated python 2-3 collision workaround, seems it's not needed anymore

Signed-off-by: Hofi <hofione@gmail.com>

HofiOne commented on .github/workflows/macos.yml:
```diff
@@ -31,7 +31,9 @@ jobs:
           PYTHONUSERBASE="${HOME}/python_packages"
-          PKG_CONFIG_PATH="${HOMEBREW_PREFIX}/opt/openssl@1.1/lib/pkgconfig:$PKG_CONFIG_PATH"
+          PKG_CONFIG_PATH="${HOMEBREW_PREFIX}/opt/openssl@1.1/lib/pkgconfig:${HOMEBREW_PREFIX}/opt/net-snmp/lib/pkgconfig:${PKG_CONFIG_PATH}"
+          CFLAGS="-I${HOMEBREW_PREFIX}/include/ ${CFLAGS}"
+          LDFLAGS="-L${HOMEBREW_PREFIX}/lib ${LDFLAGS}"
```
pkgconfig is bogus on some OS versions, unfortunately, gives back invalid paths

HofiOne replied:
forcing usage of homebrew versions is a must have, there are multiple issues if all the macOS supported modules are enabled (--enable-all-modules --disable-smtp --disable-mqtt --disable-pacct --disable-java)

HofiOne commented on .github/workflows/macos.yml:
```diff
@@ -31,7 +31,9 @@ jobs:
           HOMEBREW_PREFIX="$(brew --prefix)"
           PYTHONUSERBASE="${HOME}/python_packages"
-          PKG_CONFIG_PATH="${HOMEBREW_PREFIX}/opt/openssl@1.1/lib/pkgconfig:$PKG_CONFIG_PATH"
+          PKG_CONFIG_PATH="${HOMEBREW_PREFIX}/opt/openssl@1.1/lib/pkgconfig:${HOMEBREW_PREFIX}/opt/net-snmp/lib/pkgconfig:${PKG_CONFIG_PATH}"
```
pkgconfig is bogus on some OS versions, unfortunately, gives back invalid paths, must use the homebrew version to get correct values

MrAnno commented:
Homebrew will try to adapt to the changes we made. We just have to ensure that the `gcc/g++` dependency is optional, as clang is preferred on their side.

MrAnno commented:
> modified the Homebrew environment building, added gcc-11, and forced usage of it

What does "forced usage" mean? I can't see it in the diff.

HofiOne commented:
> 

- Added ${HOMEBREW_PREFIX}/bin to the PATH
- Added CFLAGS="-I${HOMEBREW_PREFIX}/include/ ${CFLAGS}"
- Added LDFLAGS="-L${HOMEBREW_PREFIX}/lib ${LDFLAGS}"
 
these are forcing the build flow to use the Homebrew provided tools, libs and headers (I hope so)

HofiOne commented:
Oh, you meant the gcc-11, yepp, the link and the ${HOMEBREW_PREFIX}/bin PATH addition
